aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1491,37 +1507,30 @@ void _csr_to_block_csr_cpu_kernel(
  * https://github.com/scipy/scipy/blob/8a64c938ddf1ae4c02a08d2c5e38daeb8d061d38/scipy/sparse/sparsetools/csr.h
  */
 template <class I>
-I csr_count_blocks(
-    const I n_row,
-    const I n_col,
-    const I R,
+I compressed_count_blocks(
+    const I n_compressed_dim,
+    const I n_plain_dim,
     const I C,
+    const I P,
     const I Ap[],
     const I Aj[]) {
-  std::vector<I> mask(n_col / C + 1, -1);
+  std::vector<I> mask(n_plain_dim / P + 1, -1);
   I n_blks = 0;
-  for (I i = 0; i < n_row; i++) {
-    I bi = i / R;
-    for (I jj = Ap[i]; jj < Ap[i + 1]; jj++) {
-      I bj = Aj[jj] / C;
-      if (mask[bj] != bi) {
-        mask[bj] = bi;
+  for (I c = 0; c < n_compressed_dim; c++) {
+    I bc = c / C;
+    for (I p = Ap[c]; p < Ap[c + 1]; p++) {
+      I bp = Aj[p] / P;
+      if (mask[bp] != bc) {
+        mask[bp] = bc;
         n_blks++;
       }
     }
   }
   return n_blks;
 }
 
-Tensor _csr_to_block_csr_cpu(const Tensor& self, IntArrayRef blocksize) {
-  TORCH_CHECK(
-      blocksize[0] == blocksize[1],
-      "blocks must be square. ",
-      "Got (",
-      blocksize[0],
-      ", ",
-      blocksize[1],
-      ") instead.");
+template <bool compressed_rows>
```

Comment:
Minor nit: it might be more natural to parametrize this template using `Layout layout` and reuse helper functions from aten/src/ATen/SparseCsrTensorUtils.h

Reply:
Changed.